File: 020-serpro-estudos-prova-pratica/src/especificacao-sem-bd/nivel-avancado/avancado.js
/*
Exercício: Desenvolva um programa que simule um sistema de gerenciamento de tarefas (to-do list). O programa deve permitir que o usuário adicione, marque como concluída e liste as tarefas. Utilize um banco de dados (por exemplo, SQLite ou MongoDB) para armazenar informações sobre as tarefas.
*/


const ListaDeTarefas = require("./ListaDeTarefas.js");
const BancoDeDados = require("./BancoDeDados.js")


async function main() {
	console.log("iniciando");
	const db = new BancoDeDados();
	await db.criarTabela();

	lista = new ListaDeTarefas(db);
	await lista.adicionar("t1");
	await lista.adicionar("t2");
	await lista.adicionar("t3");

	const retorno = await lista.consultar();
	console.log(retorno);

	db.fechar();
}

main();

File: 020-serpro-estudos-prova-pratica/src/especificacao-sem-bd/nivel-avancado/avancado.py
"""
Nível Avançado:

Exercício: Desenvolva um programa que permita ao usuário criar uma lista de tarefas. O programa deve permitir que o usuário adicione tarefas, marque tarefas como concluídas e exiba a lista de tarefas pendentes e concluídas.
"""

from Tarefa import Tarefa

t1 = Tarefa("task one")
t2 = Tarefa("ttwo")

lista_de_tarefas = []
lista_de_tarefas.append(t1)
lista_de_tarefas.append(t2)

print("Total de " + str(len(lista_de_tarefas)) + " tarefas.")

tRecuperado = lista_de_tarefas[1]
tRecuperado.concluir()


for chave, tarefa in enumerate(lista_de_tarefas):
	print(str(chave) + "." + tarefa.descricao())